src/vcpkg/install.cpp

```diff
@@ -1070,6 +1079,79 @@ namespace vcpkg
                 extended_overlay_ports.emplace_back(paths.builtin_ports_directory().native());
             }
 
+            std::string hash;
+            {
+                auto hasher = Hash::get_hasher_for(Hash::Algorithm::Sha256);
+                hasher->add_bytes(default_triplet.to_string());
+                hasher->add_bytes(host_triplet.to_string());
+                for (const auto& feature : features)
+                {
+                    hasher->add_bytes(feature);
+                }
+                if (paths.get_registry_set().is_default_builtin_registry())
+                {
+                    auto maybe_git_sha = paths.get_current_git_sha();
+                    if (maybe_git_sha)
```

Comment:
I'm not sure just giving up and not recording the hash if we aren't in a git repo is the right behavior. We need the git part to detect that none of the depended-upon stuff changed.